dvc/repo/experiments/queue/base.py

```diff
@@ -521,21 +522,36 @@ def _update_params(self, params: Dict[str, List[str]]):
                 provided via `exp run --set-param`.
 
         .. _Hydra Override:
-            https://hydra.cc/docs/next/advanced/override_grammar/basic/
+            https://hydra.cc/docs/advanced/override_grammar/basic/
         """
         logger.debug("Using experiment params '%s'", params)
 
         try:
-            from dvc.utils.hydra import apply_overrides
+            from dvc.utils.hydra import apply_overrides, compose_and_dump
         except ValueError:
             if sys.version_info >= (3, 11):
                 raise DvcException(
                     "--set-param is not supported in Python >= 3.11"
                 )
             raise
 
+        hydra_config = self.repo.config.get("hydra", {})
+        hydra_enabled = hydra_config.get("enabled", False)
+        hydra_output_file = ParamsDependency.DEFAULT_PARAMS_FILE
         for path, overrides in params.items():
-            apply_overrides(path, overrides)
+            if hydra_enabled and path == hydra_output_file:
+                config_dir = os.path.join(
+                    self.repo.root_dir, hydra_config.get("config_dir", "conf")
+                )
+                config_name = hydra_config.get("config_name", "config")
+                compose_and_dump(
```

Comment:
Could be worth it to check that `config_dir/config_name` exists and properly handled that case, right now if you run `exp run -v` with a non-existing config, you get the following:

> ERROR: unexpected error - Cannot find primary config 'nonexisting'. Check that it's in your config search path.
> 
> Config search path:
> 	provider=hydra, path=pkg://hydra.conf
> 	provider=main, path=file:///home/dtrifiro/dvc/hydra_conf
>	provider=schema, path=structured://
```